base/sort.jl

```diff
@@ -789,16 +789,51 @@ partialsortperm(v::AbstractVector, k::Union{Integer,OrdinalRange}; kwargs...) =
 """
     partialsortperm!(ix, v, k; by=<transform>, lt=<comparison>, rev=false, initialized=false)
 
-Like [`partialsortperm`](@ref), but accepts a preallocated index vector `ix`. If `initialized` is `false`
-(the default), `ix` is initialized to contain the values `1:length(ix)`.
-"""
+Like [`partialsortperm`](@ref), but accepts a preallocated index vector `ix` the same size as
+`v`.
+
+If the index vector `ix` is initialized with the indices of `v`, `initialized` should be set to
+`true`.
+
+If `initialized` is `false` (the default), then `ix` is initialized to contain the indices of `v`.
+
+(Typically, the indices of `v` will be `1:length(ix)`, although if `v` is an alternative array types
```

Comment:
"is of an alternative array type"? (add "of", remove "s")

Reply:
```suggestion
(Typically, the indices of `v` will be `1:length(ix)`, although if `v` is an alternative array type
```

Reply:
I'm not a native speaker, but I thought that `v` is an array, not an array type. So, isn't an array "of some (array) type"?

Reply:
Sorry, I was in too much of a hurry.  An array can be "of a type", as you suggested, or we can say it "has a type", which I think works better here.

Reply:
Just maybe `1:length(ix)` could be replaced by `1:length(v)`, just to highlight that the size of `ix` depends on the size of `v`?